addons/mrp_account/models/product.py

```diff
@@ -58,7 +58,11 @@ def _compute_average_price(self, qty_invoiced, qty_to_invoice, stock_moves, is_r
         for bom_line, moves_list in groupby(stock_moves.filtered(lambda sm: sm.state != 'cancel'), lambda sm: sm.bom_line_id):
             if bom_line not in bom_lines:
                 for move in moves_list:
-                    value += move.product_id._compute_average_price(qty_invoiced * move.product_qty, qty_to_invoice * move.product_qty, move, is_returned=is_returned)
+                    component_quantity = next(
+                        (bml.product_qty for bml in move.product_id.bom_line_ids if bml in bom_lines),
+                        1
+                    )
```

Comment:
will this get the right quantity in case of bom component of bom that I think bom.explode will manage?

if not, maybe you could use the bom line directly with a dictionary like

```py
component_quantities = {line.product_id: data['qty'] for line, data in bom_lines.items()}
```
then use it with eg. `component_quantities.get(move.product_id, 1)`